src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -222,6 +223,55 @@ export class TabsTitleControl extends TitleControl {
 		});
 	}
 
+	private updateTabSizing(): void {
+		if (this.tabsContainer) {
+			const options = this.accessor.partOptions;
+			if (options.tabSizing === 'fixed') {
+				this.tabsContainer.style.setProperty('--tab-sizing-fixed-max-width', `${options.tabSizingFixedMaxWidth}px`);
+				this.tabsContainer.onmouseenter = () => this.setTabsWidthFixed();
```

Comment:
Here I suggest to use `DisposableStore` together with `addDisposableListener`. You can create a `private readonly tabSizingFixedDisposables = this._register(new DisposableStore)` and then add the 2 listeners to it. When the setting changes to be turned off, simply call `tabSizingFixedDisposables.clear()`

Reply:
Thank you, exactly the guidance I needed. 8-)

Reply:
Sadly, the initializer would get called after the constructor, so I need to make it mutable and set it in create(). Doable.

Reply:
You can still do this, no?

```ts
private tabSizingFixedDisposables = this._register(new DisposableStore());
```

Reply:
Unless I've missed something, this doesn't work: the code that assigns this value gets called *after* the constructor; and it's the constructor which indirectly calls `create()`. Fields like this are not initialized until the constructor finishes. It might be nice to initialize it in the constructor, but again, we'd have to do it after the call to `super()`, but it's that superclass which calls our `create()`, so we must initialize it inside `create()`.

Reply:
You are right!